aws/resource_aws_s3_bucket.go

```diff
@@ -1782,9 +1782,25 @@ func resourceAwsS3BucketServerSideEncryptionConfigurationUpdate(s3conn *s3.S3, d
 	}
 	log.Printf("[DEBUG] S3 put bucket replication configuration: %#v", i)
 
-	_, err := retryOnAwsCode(s3.ErrCodeNoSuchBucket, func() (interface{}, error) {
-		return s3conn.PutBucketEncryption(i)
+	err := resource.Retry(2*time.Minute, func() *resource.RetryError {
```

Comment:
There's a [RetryOnAwsCodes](https://github.com/terraform-providers/terraform-provider-aws/blob/v2.46.0/aws/awserr.go#L41-L43) function that could be used instead of this in order to retry for both `OperationAborted` and `NoSuchBucket` errors. The comment that it "will be moved out of the aws package in the future" made me wonder if new code should be using it, though. Also, `RetryOnAwsCodes` limits the maximum retries to 1 minute and it appears that for `NoSuchBucket` errors that the code was intentionally changed at one point to retry up to 2 minutes. Also, I scoped the new code to only retry for `OperationAborted` errors for newly created resources since I'm not sure if the extra retry is needed for updated resources. This is why I'd gone with the inline block below. Happy to change this to something else, though, if there's a preferred way to do it.